src/logic/map.cc

```diff
@@ -1733,31 +1733,25 @@ int Map::calc_buildsize(const EditorGameBase& egbase,
 }
 
 /**
- * We call a cycle on the map simply connected
- * if the subgraph of the cycle which can be walked on is connected.
+ * We call a cycle on the map simply 'connected' if it can be walked on.
  *
  * The cycle is described as a \p start point plus
  * a description of the directions in which to walk from the starting point.
  * The array \p dirs must have length \p length, where \p length is
  * the length of the cycle.
  */
 bool Map::is_cycle_connected(const FCoords& start, uint32_t length, const WalkingDir* dirs) const {
+	if (!(start.field->get_caps() & MOVECAPS_WALK)) {
+		return false;
+	}
 	FCoords f = start;
-	bool prev_walkable = start.field->get_caps() & MOVECAPS_WALK;
-	uint32_t alternations = 0;
-
 	for (uint32_t i = 0; i < length; ++i) {
```

Comment:
Since we're touching this function anyway, we could modernize the loop and get rid of the "length" function argument.

```
for (WalkingDir* dir : dirs) {
    f = get_neighbour(f, dir);
```

Reply:
My compiler doesn't like it since the array is passed as a pointer function argument, so I converted it to vector instead.